uwdata/lyra
# De-couple contenteditable from signalvalue and use store to save names

**Description of the problem this pull request fixes**

The ContentEditable mixins had not been updated to persist the updated strings in the main lyra store, meaning that mark name changes were not consistently represented throughout the application.

This also cleans up the way SignalValueMixin had been overloaded to handle ContentEditable data persistence: Instead, `<ContentEditable>` now takes a `save={someFn}` attribute, and that save method is called with the updated value of the editable area.

In cases where this value is a mark name that is represtented in the store, we use `save` to dispatch an action that sets the value in the store; and for now we also set the value on the primitive itself, since primitives are only sync'd from the store when a change occurs that would warrant a Vega re-parse. (And for now I believe renaming a mark is not in that category).

If the value being updated is _not_ yet part of the store, we simply define a contextual `save` method that will set the value on whatever instantiated primitive we want to modify.

Changes proposed in this pull request:
- Add an `updateMarkProperty` action dispatcher & reducer
- De-couple ContentEditable from SignalValueMixin
- Update ContentEditable to use a save method passed in via property attribute
- Set ContentEditable components to update their values in instances or the store or both

*\* If submitting code for review: **

**This PR includes:**
- ~ high level description of any new components

**Steps to functionally test this:**
- Create a mark; rename it; observe that the name is updated in the inspector sidebar
- Create a group; create a mark within it; rename the mark; close its parent group; reopen the group; observe that the mark's name is still what you entered
- Rename a pipeline; use the `window.primitives` global in the console to observe that the name is saved to the object
- Bind data; rename a scale; use the `window.primitives` global in the console to observe that the name is saved to the object
